### PR TITLE
canbus: isotp: Fix nullptr deref and wrong variable in assert (Coverity issues)

### LINKED ISSUE #22656: [Coverity CID :208192] Out-of-bounds access in tests/subsys/canbus/isotp/implementation/src/main.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/e089906b339aad4cd1b6589a3b6ce94782d93f54/tests/subsys/canbus/isotp/implementation/src/main.c#L197

Category: Memory - corruptions
Function: `receive_test_data`
Component: Tests
CID: [208192](https://scan9.coverity.com/reports.htm#v29726/p12996/mergedDefectId=208192)

Details:

```
196             recv_len = isotp_recv(recv_ctx, data_buf, sizeof(data_buf),
197                           K_MSEC(1000));
198             zassert_true(recv_len >= 0, "recv error: %d", recv_len);
199    
200             zassert_true(remaining_len >= recv_len,
201                      "More data then expected");
>>>     CID 208192:    (OVERRUN)
>>>     Overrunning array "data_buf" of 128 bytes by passing it to a function which accesses it at byte offset 4294967286 using argument "recv_len" (which evaluates to 4294967287).
202             ret = check_data(data_buf, data_ptr, recv_len);
203             zassert_equal(ret, 0, "Data differ");
204             data_ptr += recv_len;
205             remaining_len -= recv_len;
206    
207             if (delay) {
191             memset(data_buf, 0, sizeof(data_buf));
192             ret = isotp_recv(recv_ctx, data_buf, sizeof(data_buf),
193                      K_MSEC(1000));
194             zassert_true(data_buf >= 0, "recv error: %d", ret);
195    
196             zassert_true(remaining_len >= ret, "More data then expected");
>>>     CID 208192:    (OVERRUN)
>>>     Overrunning array "data_buf" of 128 bytes by passing it to a function which accesses it at byte offset 4294967286 using argument "ret" (which evaluates to 4294967287).
197             check_data(data_buf, data_ptr, ret);
198             data_ptr += ret;
199             remaining_len -= ret;
200    
201             if (delay) {
202                 k_sleep(delay);
```

Please fix or provide comments in coverity using the link:

https://scan9.coverity.com/reports.htm#v32951/p12996.

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the CODEOWNERS file.

### LINKED ISSUE #22657: [Coverity CID :208191] Dereference after null check in subsys/canbus/isotp/isotp.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/e089906b339aad4cd1b6589a3b6ce94782d93f54/subsys/canbus/isotp/isotp.c#L437

Category: Null pointer dereferences
Function: `receive_add_mem`
Component: Other
CID: [208191](https://scan9.coverity.com/reports.htm#v29726/p12996/mergedDefectId=208191)

Details:

```
431         ctx->act_frag = ctx->act_frag->frags;
432         if (!ctx->act_frag) {
433             LOG_ERR("No fragmet left to append data");
434             receive_report_error(ctx, ISOTP_N_BUFFER_OVERFLW);
435         }
436    
>>>     CID 208191:  Null pointer dereferences  (FORWARD_NULL)
>>>     Passing null pointer "&(*ctx->act_frag).b" to "net_buf_simple_add_mem", which dereferences it.
437         net_buf_add_mem(ctx->act_frag, data + tailroom, len - tailroom);
438     }
439    
440     static void process_cf(struct isotp_recv_ctx *ctx, struct zcan_frame *frame)
441     {
442         u32_t *ud_rem_len = (u32_t *)net_buf_user_data(ctx->buf);
```

Please fix or provide comments in coverity using the link:

https://scan9.coverity.com/reports.htm#v32951/p12996.

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the CODEOWNERS file.

### PR DESCRIPTION
**Pull request by alexanderwachter**
This PR is a fix for CID 208191(#22657) and CID 208192 (#22656)

fixes #22657 
fixes #22656

### CODE REVIEW

File: subsys/canbus/isotp/isotp.c [hunk 1]
```diff
@@ -432,6 +432,7 @@ static inline void receive_add_mem(struct isotp_recv_ctx *ctx, u8_t *data,
 	if (!ctx->act_frag) {
```
Comment (jukkar) [inline subsys/canbus/isotp/isotp.c]: In commit message "canbus: isotp: Fix NULL deref after overflow", better would be to have it like this, to make these one look like in other PRs.
```
Fixes #22657
Coverity-CID: 208191
```

File: tests/subsys/canbus/isotp/implementation/src/main.c [hunk 2]
```diff
@@ -191,7 +191,7 @@ static void receive_test_data(struct isotp_recv_ctx *recv_ctx,
 		memset(data_buf, 0, sizeof(data_buf));
```
Comment (jukkar) [inline tests/subsys/canbus/isotp/implementation/src/main.c]: Same for the second patch:
```
Fixes #22656
Coverity-CID: 208192
```

Comment (alexanderwachter) [reply tests/subsys/canbus/isotp/implementation/src/main.c]: Done

File: subsys/canbus/isotp/isotp.c (hunk 1 above)
Comment (alexanderwachter) [reply subsys/canbus/isotp/isotp.c]: Done

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**